client/signup/jetpack-connect/authorize-form.jsx

```diff
@@ -478,10 +478,10 @@ const LoggedInForm = React.createClass( {
 	renderStateAction() {
 		const { authorizeSuccess, siteReceived } = this.props.jetpackConnectAuthorize;
 		if (
-			this.props.isFetchingSites ||
+			this.props.isFetchingSite ||
 			this.isAuthorizing() ||
 			this.retryingAuth ||
-			( authorizeSuccess && ! siteReceived )
+			( this.props.isFetchingSites && authorizeSuccess && ! siteReceived )
```

Comment:
It feels a little off to make this condition more strict, can you explain the reasoning?